Makefile refactor download redis

LarryBattle opened the pull request:
## Description
- Simplified the logic for the make target `work/redis-git/src/redis-server`
- Set a shell version to bash. Ubuntu was defaulting to `/bin/sh` which caused the `[ "$(REDIS)" == "unstable" ]` expression to error out


Pseudo code for changes to the make target `work/redis-git/src/redis-server`
Originally it was doing this.

```js
if( !hasRedisGitDir ){
    downloadRedisGit()
    checkoutAndPullBranch( REDIS )
}
REDIS_BRANCH = getBranchName()
gitResetBranch()
if( REDIS != REDIS_BRANCH ){
    checkoutAndPullBranch( REDIS )
}
if( REDIS == DEFAULT_REDIS_BRANCH ){
    checkoutAndPullBranch( REDIS )
}
```

Rewrote it to do this
```js
if( hasRedisGitDir ){
    gitResetBranch()
} else {
    downloadRedisGit()
}
checkoutAndPullBranch( REDIS )
```

### Test command

```bash
clear
make clean && make 'work/redis-git/src/redis-server'
make clean && make 'work/redis-git/src/redis-server' REDIS=3.2.11
make clean && make 'work/redis-git/src/redis-server' REDIS=4.0.11
```

### Test command output

```bash
[ -d "work/redis-git" ] && cd work/redis-git && git reset --hard || \
git clone https://github.com/antirez/redis.git work/redis-git
Cloning into 'work/redis-git'...
remote: Enumerating objects: 12, done.
remote: Counting objects: 100% (12/12), done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 54635 (delta 4), reused 7 (delta 4), pack-reused 54623
Receiving objects: 100% (54635/54635), 73.10 MiB | 2.76 MiB/s, done.
Resolving deltas: 100% (38472/38472), done.
cd work/redis-git && git checkout -q unstable && git pull origin unstable
From https://github.com/antirez/redis
 * branch              unstable   -> FETCH_HEAD
Already up to date.

*** NOTE: make steps remove from output

[ -d "work/redis-git" ] && cd work/redis-git && git reset --hard || \
git clone https://github.com/antirez/redis.git work/redis-git
HEAD is now at 3d07ed98 Fix typo in replicationCron() comment.
cd work/redis-git && git checkout -q 3.2.11 && git pull origin 3.2.11
From https://github.com/antirez/redis
 * tag                 3.2.11     -> FETCH_HEAD
Already up to date.

*** NOTE: make steps remove from output

[ -d "work/redis-git" ] && cd work/redis-git && git reset --hard || \
git clone https://github.com/antirez/redis.git work/redis-git
HEAD is now at 1cb6effd Redis 3.2.11.
cd work/redis-git && git checkout -q 4.0.11 && git pull origin 4.0.11
From https://github.com/antirez/redis
 * tag                 4.0.11     -> FETCH_HEAD
Already up to date.

*** NOTE: make steps remove from output
```

mp911de commented:
Pretty decent refactoring. Thanks a lot.

mp911de commented:
For some reason, redis binaries are getting built multiple times, checkout TravisCI https://travis-ci.org/lettuce-io/lettuce-core/jobs/438076234 logs.

LarryBattle commented:
I removed the `.PHONY` statement, which caused the redis binaries to get rebuilt.
The reason I added it was because you can't change the redis binaries without running `make clean`.
`make` assumes everything is updated if the first expression of a target is a passing file check.
`[ -d 'work/redis-git']`

Ex.
```
$ make 'work/redis-git/src/redis-server'
*** omitted output from building the redis binaries 

$ make 'work/redis-git/src/redis-server' REDIS=3.2.11
make: 'work/redis-git/src/redis-server' is up to date.
```

So running this only uses one version of the redis binaries and not two.

```bash
make clean
make prepare ssl-keys test
make prepare ssl-keys test REDIS=3.2.11
```

Would you like me to add a check in the make target `'work/redis-git/src/redis-server'` to rebuild the redis binaries if needed?

mp911de commented:
Thanks a lot 🙇. That's squashed, merged and backported now.
